File: serverless/src/materialsHandler.js
import { getCourseMaterials } from "./materials.js";
import httpResponse from "./http_response.js";

export const handler = async (event) => {
  try {
    return await getCourseMaterials();
  } catch (err) {
    return httpResponse(err.statusCode, err.message);
  }
};

File: serverless-python/src/materialsHandler.py
from botocore.exceptions import ClientError
from src.materials import get_course_materials
from src.http_response import create_response

def handler(event, context):
    query_params = event.get('queryStringParameters', {})
    topic = query_params.get('topic')
    try:
        return get_course_materials(topic)
    except ClientError as err:
      statusCode = err.statusCode if hasattr(err, 'statusCode') else 500
      message = err.message if hasattr(err, 'message') else 'Internal Server Error'
      return create_response(statusCode, message)

